mock_drivers/rosbag_mock_drivers/README.md

```diff
@@ -0,0 +1,108 @@
+# rosbag_mock_drivers
+
+This package contains a node which can mimic the behavior of different types of drivers in the CARMA Platform.
+These fake drivers are refereed to as mock_drivers.
+The mock drivers in this package function using real rosbag data.
+The general behavior is that the rosbag is played under the ```/bag/``` prefix and
+then the node will serve as a relay between that bag topic and the CARMA Platform expected topic.
+If the message type included a header, then the timestamp will be updated to reflect the ros::Time::now().
+This allows the nodes to function both with and without the usage of the ```--clock``` option for the rosbag playback. For example, the Lidar mock driver has the following behavior:
+
+* It subscribes to the ```/bag/hardware_interface/lidar/points_raw``` topic.
+* It publishes to the ```/hardware_interface/lidar/points_raw``` topic.
+* When a message is received from the bag it is forwarded with an updated timestamp.
+* Status heartbeats are sent on the ```/hardware_interface/driver_discovery``` at an ~1 Hz frequency.
+
+## Input Data
+
+The drivers require a rosbag located at the ```/opt/carma/data/mock_drivers/drivers.bag``` file location in order to properly function.
+This file is expected to contain all topics expected by the mock drivers which are being run.
+
+## Creating new mock drivers
+
+If a developer wants to implement a new mock driver they should extend the MockDriver class found in the ```include/rosbag_mock_drivers/MockDriver.h``` file. There new class should be located in ```include/rosbag_mock_drivers/``` This would give them something similar to the following:
+
```

Comment:
The file name is README.md and not a .h file.

Reply:
I'm not quite following. This is the package README file not a code file which is why it does not contain use .h extension. It's located at the package root.